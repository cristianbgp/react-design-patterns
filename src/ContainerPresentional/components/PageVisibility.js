import React from "react";

function PageVisibility({ status }) {
  return (
    <p>
      Page Visibility: <span>{status.toString()}</span>
    </p>
  );
}

export default PageVisibility;
