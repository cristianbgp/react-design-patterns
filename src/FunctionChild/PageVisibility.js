import React from "react";

function PageVisibility({ children }) {
  const [status, setStatus] = React.useState(true);

  React.useEffect(() => {
    function handleVisibility(event) {
      if (document.hidden) {
        console.log("Page is not visible");
        setStatus(false);
      } else {
        console.log("Page is visible");
        setStatus(true);
      }
    }
    document.addEventListener("visibilitychange", handleVisibility);
    return () => {
      window.removeEventListener("visibilitychange", handleVisibility);
    };
  }, [setStatus]);

  return children(status);
}

export default PageVisibility;
