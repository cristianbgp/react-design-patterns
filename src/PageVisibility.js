import React from "react";

function PageVisibility({ children, render = children }) {
  const [status, setStatus] = React.useState(true);

  function handleVisibility(event) {
    if (document.hidden) {
      console.log("Page is not visible");
      setStatus(false);
    } else {
      console.log("Page is visible");
      setStatus(true);
    }
  }

  React.useEffect(() => {
    document.addEventListener("visibilitychange", handleVisibility);
    return () => {
      window.removeEventListener("visibilitychange", handleVisibility);
    };
  }, [setStatus]);

  return render(status);
}

export default PageVisibility;
