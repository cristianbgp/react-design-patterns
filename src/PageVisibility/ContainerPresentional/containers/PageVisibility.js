import React from "react";
import PageVisibility from "../components/PageVisibility";

function PageVisibilityContainer() {
  const [status, setStatus] = React.useState(true);

  React.useEffect(() => {
    function handleVisibility(event) {
      if (document.hidden) {
        console.log("Page is not visible");
        setStatus(false);
      } else {
        console.log("Page is visible");
        setStatus(true);
      }
    }
    document.addEventListener("visibilitychange", handleVisibility);
    return () => {
      window.removeEventListener("visibilitychange", handleVisibility);
    };
  }, [setStatus]);

  return <PageVisibility status={status} />;
}

export default PageVisibilityContainer;
