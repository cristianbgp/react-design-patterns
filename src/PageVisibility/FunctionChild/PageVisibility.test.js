import React from "react";
import { render } from "@testing-library/react";

import PageVisibility from "./PageVisibility";

test("Page Visibility with function as children", () => {
  function Component() {
    return (
      <>
        <h1>Using Function as children</h1>
        <PageVisibility>
          {status => {
            return (
              <p>
                Page Visibility: <span>{status.toString()}</span>
              </p>
            );
          }}
        </PageVisibility>
      </>
    );
  }

  const { asFragment } = render(<Component />);

  expect(asFragment()).toMatchSnapshot();
});
