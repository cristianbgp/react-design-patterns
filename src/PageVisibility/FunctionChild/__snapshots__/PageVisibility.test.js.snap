// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Visibility with function as children 1`] = `
<DocumentFragment>
  <h1>
    Using Function as children
  </h1>
  <p>
    Page Visibility: 
    <span>
      true
    </span>
  </p>
</DocumentFragment>
`;
